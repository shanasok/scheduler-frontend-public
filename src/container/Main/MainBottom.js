import React from 'react';

export class MainBottom extends React.Component {

    render() {
        return <div id="mainBottom">
            Main Bottom goes here
        </div>
    }
}