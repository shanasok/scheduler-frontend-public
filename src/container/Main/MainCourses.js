import React from 'react';
import {CourseOption} from "../CourseOption/CourseOption";

export function populateCourseOption(courses){
    if (courses && Array.isArray(courses)){
        return courses.map(course =>
            <CourseOption key={course.id} course={course}/>
        )
    }
}


export default class MainCourses extends React.Component {

    render() {
        const courseOptions = populateCourseOption(this.props.courses);

        return <div id="mainCourses">
            {courseOptions}
        </div>
    }
}