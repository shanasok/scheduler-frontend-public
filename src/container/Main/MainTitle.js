import React from 'react';

export class MainTitle extends React.Component {

    render() {
        return <div id="mainTitle">
            Click to see course content, materials and scheduling information.
        </div>
    }
}