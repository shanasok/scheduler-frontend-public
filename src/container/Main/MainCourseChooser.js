import React from 'react';
import { connect } from "react-redux";

import {MainBottom} from "./MainBottom";
import MainCourses from "./MainCourses";
import {MainTitle} from "./MainTitle";

import '../../css/CourseOptionStyles.css';

const MainCourseChooser = props => (
    <div id="mainCourseChooser">
        <MainTitle />
        <MainCourses courses={props.courses} />
        <MainBottom />
    </div>
);

const mapStateToProps = state => ({
    courses: state.courses,
});

export default connect(mapStateToProps)(MainCourseChooser)