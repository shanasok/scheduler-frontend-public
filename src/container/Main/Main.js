import React from 'react';
import {Switch, Route, BrowserRouter as Router} from "react-router-dom";
import MainCourseChooser from "./MainCourseChooser";
import CourseDetails from "../courseBreakdown/CourseDetails"
import LessonSlide from "../courseBreakdown/subjectBreakdown/lessonBreakdown/LessonSlide";
import NotFound from "../NotFound";

export class Main extends React.Component {

    render() {
        return <div id="main">
            <Router>
                <Switch>
                    <Route exact path="/" component={() => <MainCourseChooser />} />
                    <Route exact path="/course" component={() => <CourseDetails />} />
                    <Route exact path="/lesson" component={() => <LessonSlide />} />
                    <Route component={() => <NotFound/>} />
                </Switch>
            </Router>
        </div>
    }
}