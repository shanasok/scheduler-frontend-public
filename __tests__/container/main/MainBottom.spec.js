import React from 'react';
import {MainBottom} from '../../../src/container/Main/MainBottom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
        .create(<MainBottom/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});