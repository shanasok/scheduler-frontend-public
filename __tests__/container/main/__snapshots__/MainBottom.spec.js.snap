// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="mainBottom"
>
  Main Bottom goes here
</div>
`;
